clients/venice-thin-client/src/main/java/com/linkedin/venice/client/schema/RouterBasedStoreSchemaFetcher.java

```diff
@@ -74,6 +74,33 @@ public Schema getLatestValueSchema() {
     }
   }
 
+  @Override
+  public Schema getSupersetSchema() {
+    String valueSchemaRequestPath = TYPE_VALUE_SCHEMA + "/" + storeClient.getStoreName();
+    MultiSchemaResponse multiSchemaResponse = fetchAllValueSchemas(valueSchemaRequestPath);
+    int supersetSchemaId = multiSchemaResponse.getSuperSetSchemaId();
+    if (supersetSchemaId == SchemaData.INVALID_VALUE_SCHEMA_ID) {
+      return null;
+    }
+    String schemaStr = null;
+    for (MultiSchemaResponse.Schema schema: multiSchemaResponse.getSchemas()) {
+      if (supersetSchemaId == schema.getId()) {
+        schemaStr = schema.getSchemaStr();
+      }
```

Comment:
If we keep using this multi-schema endpoint and keep this loop here... then I guess we can `break` once we find the right schema?

Also, are the schemas in that collection jumbled up or in some kind of order? Can we just get the one we need by looking up the right index...?